personhood/contract_spawner.go

```diff
@@ -80,25 +82,46 @@ func (c *ContractSpawner) Spawn(rst byzcoin.ReadOnlyStateTrie, inst byzcoin.Inst
 	case ContractSpawnerID:
 		c.CostCWrite = &byzcoin.Coin{}
 		c.CostCRead = &byzcoin.Coin{}
+		c.CostValue = &byzcoin.Coin{}
 		err = c.parseArgs(inst.Spawn.Args)
 		if err != nil {
-			return nil, nil, err
+			return nil, nil, errors.New("couldn't parse args: " + err.Error())
 		}
 		instBuf, err = protobuf.Encode(&c.SpawnerStruct)
 		if err != nil {
 			return nil, nil, errors.New("couldn't encode SpawnerInstance: " + err.Error())
 		}
+
 	case byzcoin.ContractDarcID:
 		if err = c.getCoins(cout, c.CostDarc); err != nil {
 			return
 		}
 		instBuf = inst.Spawn.Args.Search("darc")
 		d, err := darc.NewFromProtobuf(instBuf)
 		if err != nil {
-			return nil, nil, err
+			return nil, nil, errors.New("couldn't decode darc: " + err.Error())
+		}
+		// Whitelist allowed darc rules - there is no spawn allowed, and all
```

Comment:
I really don't understand this. Why is this needed now? Was it wrong before? Is this something about the value contract?

Reply:
The spawner-contract allows any user with coins to create new instances. In order not to have a user create a darc with rules like `spawn:darc` or `invoke:coin.mint`, the allowed rules are restricted.

Chosing between a blacklist and whitelist approach I chose the latter, to avoid that somebody adds an `invoke` to any contract and then opens a security hole in the spawner-contract.

If you prefer, I can create a separate PR for that 'fix' of the spawner-contract.